### ISSUE TITLE
Allow for cli enqueuing of jobs with non-string arguments

### ISSUE THREAD

**Issue by kralos** [mmucklo/DtcQueueBundle#82]
Given the following job:
```php
class MyJob extends \Dtc\QueueBundle\Model\Worker
{
    public function getName()
    {
        return 'my-job';
    }

    public function run(int $someId, ?int $someOtherId): int
    {
        // Do Things
        return self::RESULT_SUCCESS;
    }
}
```

It is NOT currently possible to manually enqueue one of these via cli with:

```sh
bin/console dtc:queue:create_job my-job run 1234 null
```

as the args are interpreted as:
```php
$someId = (string) "1234";
$someOtherId = (string) "null";
```

A solution to this would be to allow a single json encoded string to be passed as the args to the cli.

Please see https://github.com/mmucklo/DtcQueueBundle/pull/83

**Comment by kralos** [comment]
@mmucklo This is a draft, please don't merge yet. I'd like to know what you think / what others think of this idea first and I might have some more testing / improvement to do to the pull request.

**Comment by mmucklo** [comment]
@kralos thus sounds like a reasonable idea.

I think maybe the standard args should be continued as strings until 5.0, however the option for json args and perhaps php serialized args could be added similar to what you have.

**Comment by kralos** [comment]
The way I've implemented it in https://github.com/mmucklo/DtcQueueBundle/pull/83 there's a new optional option `--json-args` of `-j` which turns on json args mode. NOT using it is deprecated as it will become the default in `v5.x`.

So anyone currently using the old style will still be able to use the old style.

**Comment by kralos** [comment]
I also thought about using PHP serialized args but they're a lot harder for a human to write on the cli. I think it's acceptable to expect a human to type out json. Usually I find manually enqueuing something I would do after something has gone wrong in my application (an emergency/recovery where a human is typing a command).

**Comment by kralos** [comment]
@mmucklo I think this is now mergable however it's failing CI because of an unrelated issue with composer in php `7.0`

**Comment by mmucklo** [comment]
@kralos, let me see if I can get a chance to review it in a little bit.  I may have a change or two to propose.

**Comment by kralos** [comment]
ok i've simplified `travis.yml` and fixed the issue with composer install on php 7.0

**Comment by mmucklo** [comment]
I made some comments on the changes.  Thanks for all your work on this btw.

I do thing in general it’s going to be a good enhancement.

**Comment by mmucklo** [comment]
@kralos - Should be merged and everything addressed in 4.8.0.

Please review and confirm.
